pkg/v4/resource/chartconfig/update.go

```diff
@@ -107,3 +100,43 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 
 	return chartConfigsToUpdate, nil
 }
+
+func (r *Resource) updateChartConfig(ctx context.Context, updatedChartConfig *v1alpha1.ChartConfig, guestG8sClient versioned.Interface) error {
+	chartConfigName := updatedChartConfig.ObjectMeta.Name
+
+	chartConfig, err := guestG8sClient.CoreV1alpha1().ChartConfigs(resourceNamespace).Get(chartConfigName, metav1.GetOptions{})
+	if guest.IsAPINotAvailable(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "guest cluster is not available.")
+
+		// Guest API is not available. We will retry on next reconciliation loop.
+		resourcecanceledcontext.SetCanceled(ctx)
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource reconciliation for custom object")
+
+		return nil
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// Update current CR with new desired state.
+	chartConfig.ObjectMeta.Labels = updatedChartConfig.ObjectMeta.Labels
+	chartConfig.Spec = updatedChartConfig.Spec
+
+	_, err = guestG8sClient.CoreV1alpha1().ChartConfigs(resourceNamespace).Update(chartConfig)
+	if guest.IsAPINotAvailable(err) {
```

Comment:
Above is already the check for the API availability. When we got to here it means it is available. So IMO we can remove the check here.

Reply:
Yes the 2nd check is overkill. Removed.